optimizer/plan/refiner.go

```diff
@@ -122,12 +125,35 @@ func (r *refiner) buildIndexRange(p *IndexScan) {
 	return
 }
 
+func (r *refiner) buildTableRange(p *TableScan) {
+	var pkHandleColumn *model.ColumnInfo
+	for _, colInfo := range p.Table.Columns {
+		if mysql.HasPriKeyFlag(colInfo.Flag) && p.Table.PKIsHandle {
+			pkHandleColumn = colInfo
+		}
+	}
+	if pkHandleColumn == nil {
+		return
+	}
+	rb := rangeBuilder{}
```

Comment:
If we are not holding state in rangeBuilder, i think it is better to use a global variable?